alerting/alerter-war/src/main/java/org/hawkular/metrics/alerter/groups/GroupTriggerManager.java

```diff
@@ -0,0 +1,597 @@
+            Map<String, String> memberTags = new HashMap<>();
+            memberTags.put("HawkularMetrics", "MemberTrigger");
+
+            Set<GroupMemberInfo> members = new HashSet<>();
+            for (Entry<List<String>, Map<String, List<String>>> sourceMapEntry : sourceMap.entrySet()) {
+                List<String> source = sourceMapEntry.getKey();
+                Map<String, List<String>> metricsMap = sourceMapEntry.getValue();
+
+                // If we can't form a tuple because 1 or more required dataIds have no metrics, then continue
+                if (metricsMap.size() < dataIds.size()) {
+                    Logger log = Logger.getLogger(GroupTriggerManager.MGTRunner.class);
+                    log.warnf("No members: source: %s tags: %s dataIds: %s", source, metricsMap.keySet(),
+                            dataIds);
+                    continue;
+                }
+
+                List<Map<String, String>> memberDataIdMaps = new ArrayList<>();
+
+                for (Entry<String, List<String>> metricsMapEntry : metricsMap.entrySet()) {
+                    String dataId = metricsMapEntry.getKey();
+                    List<String> metricNames = metricsMapEntry.getValue();
+
+                    if (memberDataIdMaps.isEmpty()) {
+                        for (String metricName : metricNames) {
+                            Map<String, String> memberDataIdMap = new HashMap<>();
+                            memberDataIdMap.put(dataId, metricName);
+                            memberDataIdMaps.add(memberDataIdMap);
+                        }
+                    } else {
+                        List<Map<String, String>> prevMemberDataIdMaps = memberDataIdMaps;
+                        memberDataIdMaps = new ArrayList<>();
+                        for (Map<String, String> prevMemberDataIdMap : prevMemberDataIdMaps) {
+                            for (String metricName : metricNames) {
+                                Map<String, String> memberDataIdMap = new HashMap<>();
+                                memberDataIdMap.putAll(prevMemberDataIdMap);
+                                memberDataIdMap.put(dataId, metricName);
+                                memberDataIdMaps.add(memberDataIdMap);
+                            }
+                        }
+                    }
+                }
+
+                Map<String, String> memberContext = new HashMap<>();
+                memberContext.put("source", source.toString());
+
+                for (Map<String, String> memberDataIdMap : memberDataIdMaps) {
+                    // use the memberDataIdMap hashcode as the member trigger id. We'll use this id to
+                    // determine whether the member already exists.
+                    GroupMemberInfo member = new GroupMemberInfo(mgt.getId(),
+                            String.valueOf(memberDataIdMap.hashCode()), null, null, memberContext,
```

Comment:
Another comment at this point, I don't see bad to suply a name and description, as this info is used by actions plugins to show where an alert is generated. If not, an email plugin will show undesired null/empty fields.